background/common.js

```diff
@@ -0,0 +1,33 @@
+/* global API */// msg.js
+'use strict';
+
+/**
+ * Common stuff that's loaded first so it's immediately available to all background scripts
+ */
+
+/* exported
+  addAPI
+  bgReady
+  compareRevision
+*/
+
+// Using a `var` to expose it on `window`
+// eslint-disable-next-line no-var
+var bgReady = {};
+bgReady.styles = new Promise(r => (bgReady._resolveStyles = r));
+bgReady.all = new Promise(r => (bgReady._resolveAll = r));
```

Comment:
If you prefer to separate the implementation and the declaration of bgReady, It might be better to make it an event service so we can use it like:
```js
// background.js
Promise.all([
  bgReady.once("StylesLoaded"),
  ...
])
  .then(() => {
    bgReady.emit("AllLoaded");
  })
  
// other-services.js
bgReady.once("AllLoaded")
  .then(()=> {
    // ...
  });
```

Reply:
This is a standard Promise, with a resolver stored separately so it can be called to resolve the Promise. Pretty self-evident. I don't want the event emitter because it won't provide autocompletion and "find usages" in IDE.

Reply:
I was referring to the meaning of the code. Take me as an example, I didn't know what they do until I read `content-script.js` and realized that since the init logic was moved to a separate file, you need an event center to control the load order.

If you like to keep it in this way, I suggest:

1. Add some comments.
2. Remove the underscore so we know it works like a deferred and will be called by other services.

Reply:
It's okay to use "Find usages" to verify what happens, not a problem. I'd say that 90% of code I see for the first time in any given project doesn't say much to me until I check how it's used in other places. It's perfectly normal. Adding or removing underscores doesn't improve or hinder it either. Judging by the fact that you focus on this trivial thing I guess there are no bigger issues :-)

Reply:
> Judging by the fact that you focus on this trivial thing I guess there are no bigger issues

These comments are only about the code style since it's a quick view. I didn't even realize some new lines are old code. I suggest not to rely on quick overview to find out bigger/logic/functional issues.

Reply:
There should be no functional changes in this PR, it's mostly async'ifying, splitting, moving of stuff like sorting or renaming methods.